tests/jballerina-unit-test/src/test/resources/test-src/types/xml/xml-attribute-access-syntax.bal

```diff
@@ -84,6 +84,90 @@ function testXMLAfterRemoveAttribute() {
     assert(element.toString(), "<bookStore xmlns=\"http://sample.com/test\" status=\"online\"/>");
 }
 
+function testXMLIndexedOptionalAttributeAccess() {
```

Comment:
Can we add some error tests too? Evaluating to error along the way.

Reply:
Errors are generated on attribute access when the value used for attribute access is an `xml:Comment`, `xml:ProcessingInstruction`, `xml:Text` or a `xml<xml:Element>` (element sequence). Since xml indexed access cannot result in a xml element sequence added test cases for the rest of three possible scenarios.